### ISSUE TITLE
RateLimiter does not work with IPv6 remote address

### ISSUE THREAD

**Issue by AlexanderS** [overleaf/overleaf#1282]
## Steps to Reproduce

I would like to self-host overleaf with an IPv6 connection. If the remote IP contains an IPv6 address the RateLimiter fails to generate a subnet and no login is possible:

```
{"name":"web","hostname":"overleaf","pid":6914,"level":50,"err":{"message":"Cannot generate subnet key as the ip address is not of the expected format.","name":"OError","stack":"OError: Cannot generate subnet key as the ip address is not of the expected format.\n    at RateLimiter.getSubnetKeyFromIp (/overleaf/services/web/app/src/infrastructure/RateLimiter.js:107:13)\n    at RateLimiter.consume (/overleaf/services/web/app/src/infrastructure/RateLimiter.js:71:30)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","info":{"ip":"2a01:c23:648d:6001:8712:1a3a:59eb:a3ae"}},"responseTimeMs":37,"req":{"method":"POST","url":"/login","remoteAddress":"2a01:c23:648d:6001:8712:1a3a:59eb:a3ae","headers":{"referer":"...","user-agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36","content-length":"125"}},"res":{"statusCode":500,"headers":{"content-length":"786"}},"msg":"POST /login","time":"2024-10-26T17:37:29.002Z","v":0}
```

(Same message with some line breaks:)
```
OError: Cannot generate subnet key as the ip address is not of the expected format.
    at RateLimiter.getSubnetKeyFromIp (/overleaf/services/web/app/src/infrastructure/RateLimiter.js:107:13)
    at RateLimiter.consume (/overleaf/services/web/app/src/infrastructure/RateLimiter.js:71:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

## Expected Behaviour

For IPv6 maybe the /64 should be used as subnet.

## Observed Behaviour

This regular expression does not match: https://github.com/overleaf/overleaf/blob/main/services/web/app/src/infrastructure/RateLimiter.js#L106

**Comment by mserranom** [comment]
This should be fixed in the `5.2.1` version (See https://github.com/overleaf/overleaf/issues/1281). 

The fix hasn't been merged yet to this repo, should happen very soon.

**Comment by das7pad** [comment]
Thanks again for the report.

https://github.com/overleaf/overleaf/commit/27c2e8b9384d10f808f5640eaa0d95d84a9980dd has been synced now. Closing.
